app/views/onboarding/OnboardingLocations.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useContext } from 'react';
+import PermissionsContext, { PermissionStatus } from '../../PermissionsContext';
+
+import { Icons, Images } from '../../assets';
+
+import OnboardingTemplate from './OnboardingTemplate';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import onboardingCompleteAction from '../../store/actions/onboardingCompleteAction';
+import { SetStoreData } from '../../helpers/General';
+import { PARTICIPATE } from '../../constants/storage';
+
+const OnboardingLocations = (): JSX.Element => {
+  const { authSubscription, location } = useContext(PermissionsContext);
+  const requestLocationAccess = async () => {
+    await location.request();
+    await authSubscription.request();
+    completeOnboarding();
+  };
+
+  const dispatch = useDispatch();
+  const completeOnboarding = () => {
+    // TODO: investigate if this is broken. This is directly copied from our old OnboardingPermissions, but the types look wrong.
+    const storeData = ((location.status ===
+      PermissionStatus.GRANTED) as unknown) as string;
```

Comment:
```suggestion
    const storeData = JSON.stringify(location.status === PermissionStatus.GRANTED);
```
I think the odd types were just added during the conversion to TypeScript. The `SetStoreData` method specifies that its parameters must be strings though it also calls `JSON.stringify` on any arguments that aren't strings